utils/conditions/experiments.go

```diff
@@ -170,3 +178,43 @@ func ExperimentRunning(experiment *v1alpha1.Experiment) bool {
 	passedDuration, _ := experimentutil.PassedDurations(experiment)
 	return experiment.Status.AvailableAt != nil && !passedDuration
 }
+
+func newInvalidSpecExperimentCondition(prevCond *v1alpha1.ExperimentCondition, reason string, message string) *v1alpha1.ExperimentCondition {
+	if prevCond != nil && prevCond.Message == message {
+		prevCond.LastUpdateTime = metav1.Now()
+		return prevCond
+	}
+	return NewExperimentConditions(v1alpha1.InvalidExperimentSpec, corev1.ConditionTrue, reason, message)
+}
+
+// VerifyExperimentSpec Checks for a valid spec otherwise returns a invalidSpec condition.
+func VerifyExperimentSpec(experiment *v1alpha1.Experiment, prevCond *v1alpha1.ExperimentCondition) *v1alpha1.ExperimentCondition {
+	templateNameSet := make(map[string]bool)
+	for i := range experiment.Spec.Templates {
+		template := experiment.Spec.Templates[i]
+		if template.Selector == nil {
+			message := fmt.Sprintf(MissingFieldMessage, ".Spec.Templates[].Selector")
+			return newInvalidSpecExperimentCondition(prevCond, InvalidSpecReason, message)
+		}
+
+		everything := metav1.LabelSelector{}
+		if reflect.DeepEqual(template.Selector, &everything) {
+			return newInvalidSpecExperimentCondition(prevCond, InvalidSpecReason, SelectAllMessage)
+		}
+
+		if template.MinReadySeconds > defaults.GetExperimentProgressDeadlineSecondsOrDefault(experiment) {
+			return newInvalidSpecExperimentCondition(prevCond, InvalidSpecReason, MinReadyLongerThanDeadlineMessage)
+		}
+
+		if template.Name == "" {
+			message := fmt.Sprintf(ExperimentTemplateNameEmpty, experiment.Name)
+			return newInvalidSpecExperimentCondition(prevCond, InvalidSpecReason, message)
+		}
+		if ok := templateNameSet[template.Name]; ok {
+			message := fmt.Sprintf(ExperimentTemplateNameRepeatedMessage, experiment.Name, template.Name)
+			return newInvalidSpecExperimentCondition(prevCond, InvalidSpecReason, message)
+		}
+		templateNameSet[template.Name] = true
```

Comment:
I think these messages should include the index of the template, otherwise it's hard to trace down which template has the issue.

Reply:
done!